program/src/state.rs

```diff
@@ -1102,7 +1103,32 @@ impl HealthCache {
         };
 
         let h = self.health[health_type as usize].ok_or(throw!())?;
-        Ok(h + curr_perp_health - prev_perp_health)
+
+        // Apply taker fees; Assume no referrer
+        let taker_fees = if taker_quote != 0 {
+            let taker_quote_native =
+                I80F48::from_num(info.quote_lot_size.checked_mul(taker_quote.abs()).unwrap());
+            match mango_group.find_token_index(&mngo_token::id()) {
+                None => info.taker_fee * taker_quote_native,
+                Some(mngo_index) => {
+                    let mngo_cache = &mango_cache.root_bank_cache[mngo_index];
+                    let mngo_deposits = mango_account.get_native_deposit(mngo_cache, mngo_index)?;
+                    let ref_mngo_req = I80F48::from_num(mango_group.ref_mngo_required);
+                    if mngo_deposits >= ref_mngo_req {
+                        info.taker_fee * taker_quote_native
+                    } else {
+                        // done in two steps to perfectly simulate what happens in `apply_taker_fees`
+                        info.taker_fee * taker_quote_native
```

Comment:
minor: all three branches contain `info.taker_fee * taker_quote_native`, could be moved out